src/org/ods/quickstarter/Context.groovy

```diff
@@ -145,4 +157,22 @@ class Context implements IContext {
         config.odsGitRef
     }
 
+    @NonCPS
+    String getAppDomain() {
+        config.appDomain
+    }
+
+    void setAppDomain() {
+        if (config.projectId) {
+            logger.startClocked("${config.componentId}-get-oc-app-domain")
+
+            this.openShiftService = new OpenShiftService(steps, logger)
```

Comment:
Context is just a regular bean, providing getters to access the information inside the config map.
It may be acceptable to add a setter, but not to have the setter perform complex business logic to retrieve the app domain.
The setter should get a String as parameter and just assign it to the config. The logic to provide the value should be somewhere else. Please, contact me to discuss this in detail.

Reply:
cleaned